README.md

```diff
@@ -0,0 +1,265 @@
+
+```sh
+tp-worker-server
+...
+< Logged Output >
+...
+```
+
+### Client Pings Server
+
+```sh
+tp-worker-client ping
+Server OK at 127.0.0.1:8080
+```
+
+### Client Starts a Job
+`ls -lha` used as example.
+
+The `--name` argument can be passed to provide a unique name to the job,
+otherwise we assign the job a unique name based on the command and an increment.
+Names must start with a letter to prevent confusion with pid's elsewhere. Names
+are global. Fails if the name is already taken.
+
+The `-f` or `--follow` argument is equivalent to `tail -f`, which immediately
+attaches to the streaming output of the job.
+
+```sh
+tp-worker-client start 'ls -lha'
+Server 127.0.0.1:8080: Started job 'ls1' at pid 12345: ls -lha
+```
+
+### Client Stops a Job
+
+Either a name or pid is required. Since a name must start with a letter we
+should be able to do this without requiring either `--name` or `--pid`
+arguments, though they could be available to make people feel better.
+
+The `-s` or `--signal` argument is equivalent to `kill -s`, where the user may
+select the signal to be sent to the process. Can accept either the signal
+name or number, such as `SIGKILL` or equivalently `9`. Defaults to `SIGTERM`
+or `15`. See `Signal numbering for standard signals` in `signal(7)` manpage
+for the full list of signals.
+
+```sh
+tp-worker-client stop ls1
+
+# or
+
+tp-worker-client stop 12345
+Server 127.0.0.1:8080: Stopped job 'ls1' at pid 12345: ls -lha
+```
+
+### Client Checks Job Status
+
+Possible statuses:
+* Queued (after queue tracking has been added to scope)
+* Started (the job has been sent to the system, but no output received yet)
+* Running (the job is running and output is being received)
+* Hung (the job is running, but no new output has been received for some time)
+* RunExited (the job stopped normally)
+* RunKilled (the job was killed by the user)
+* RunFailed (the job stopped abnormally)
+* Error (the job could not be run due to a system issue)
+
+The below status command prints a table of all commands run by the user since
+server start. Can also be run with multiple `--name` or `--pid` arguments to
+get a table of statuses of selected jobs submitted by the user.
+
+```sh
+tp-worker-client status
+PID   NAME  STATUS  OUTLEN SUBMITTED STARTED  STOPPED   DURATION
+12345 ls1   Running 6      15:55:55  15:55:55 15:55:57  2s
+```
+
+### Client Streams Output
+
+Requires either a name or pid to be passed, like `stop` commmand does.
+
+```
+tp-worker-client output ls1
+...
+< Streamed Output >
+...
+```
+
+## Further Development / Product Ideas
+
+These are being outlined to provide potential future directions for the product,
+a place to consider where it might be appropriate to provide Interfaces rather
+than full implementations in the libraries, and anti-priorities (work that
+should be avoided at all costs, for now) for completing the task as outlined
+above.
+
+### Scanning and Obfuscating Output
```

Comment:
This whole section is interesting, but remember the project requirement to not make any assumptions about the data. We can't assume there are any _words_ at all.

Reply:
Yep, and that's part of why this is an out-of-scope anti-priority.